frontend/src/features/mapSideBar/MotiInventoryContainer.tsx

```diff
@@ -34,163 +29,180 @@ export interface IMotiInventoryContainerProps {
-  useEffect(() => {
-    const func = async () => {
-      try {
-        if (!!props.pid && !!props.readOnly) {
-          const propInfo = await getPropertyWithPid(props.pid);
-          if (isMounted() && propInfo.pid === pidFormatter(props.pid)) {
-            setApiProperty(propInfo);
-            setShowPropertyInfoTab(true);
-          }
-        }
-      } catch (e) {
-        // PSP-2919 Hide the property info tab for non-inventory properties
-        // We get an error because PID is not on our database
-        if (axios.isAxiosError(e)) {
-          const axiosError = e as AxiosError<IApiError>;
-          if (axiosError?.response?.status === 404) {
-            setShowPropertyInfoTab(false);
-            setActiveTab(InventoryTabNames.title);
-          }
-        }
-      }
-    };
-
-    func();
-  }, [getPropertyWithPid, isMounted, props.pid, props.readOnly]);
 
   const {
     getPropertyAssociations,
     isLoading: propertyAssociationsLoading,
   } = usePropertyAssociations();
 
+  const { getLtsaData, ltsaLoading } = useLtsa();
+
+  const [composedProperty, setComposedProperty] = useState<ComposedProperty>({
+    apiPropertyLoading: ltsaLoading,
+    ltsaLoading: propertyLoading,
+    propertyAssociationsLoading: propertyAssociationsLoading,
+  });
+
+  useEffect(() => {
+    setComposedProperty(property => ({
+      ...property,
+      pid: props.pid,
+    }));
+    setIsEditing(false);
+  }, [props.pid]);
+
+  useEffect(() => {
+    setComposedProperty(property => ({
+      ...property,
+      ltsaLoading: ltsaLoading,
+      apiPropertyLoading: propertyLoading,
+      propertyAssociationsLoading: propertyAssociationsLoading,
+    }));
+  }, [propertyLoading, ltsaLoading, propertyAssociationsLoading]);
+
+  const fetchPimsProperty = React.useCallback(async () => {
+    try {
+      if (!!props.pid) {
+        const propInfo = await getPropertyWithPid(props.pid);
+        if (isMounted() && propInfo.pid === pidFormatter(props.pid)) {
+          setComposedProperty(property => ({ ...property, apiProperty: propInfo }));
+        }
+      }
+    } catch (e) {
+      // PSP-2919 Hide the property info tab for non-inventory properties
+      // We get an error because PID is not on our database
+      if (axios.isAxiosError(e)) {
+        const axiosError = e as AxiosError<IApiError>;
+        if (axiosError?.response?.status === 404) {
+          setComposedProperty(property => ({ ...property, apiProperty: undefined }));
+        }
+      }
+    }
+  }, [getPropertyWithPid, isMounted, props.pid]);
+
   useEffect(() => {
-    async function fetchResearchFile() {
+    fetchPimsProperty();
+  }, [fetchPimsProperty]);
+
+  useEffect(() => {
+    const getAssociations = async () => {
       if (props.pid !== undefined) {
         const response = await getPropertyAssociations(props.pid);
-        setPropertyAssociations(response);
+        if (response !== undefined) {
```

Comment:
if the response was undefined would we want to maintain the old response or clear it?